wrappers/desktop_test.go

```diff
@@ -99,10 +99,51 @@ func (s *desktopSuite) TestEnsurePackageDesktopFiles(c *C) {
 	sanitizedDesktopFileContent := wrappers.SanitizeDesktopFile(info, expectedDesktopFilePath, mockDesktopFile)
 	c.Check(expectedDesktopFilePath, testutil.FileContains, sanitizedDesktopFileContent)
 
-	// Old desktop file should be removed
+	// Old desktop file should be removed because it follows the
+	// <desktop-prefix>_*.desktop pattern.
 	c.Assert(osutil.FileExists(oldDesktopFilePath), Equals, false)
 }
 
+func (s *desktopSuite) TestEnsurePackageDesktopFilesWithDesktopIDs(c *C) {
+	var desktopAppYaml = `
+name: foo
+version: 1.0
+plugs:
+  desktop:
+    desktop-file-ids: [org.example.Foo]
+`
+	info := snaptest.MockSnap(c, desktopAppYaml, &snap.SideInfo{Revision: snap.R(11)})
+	c.Assert(info.Plugs["desktop"], NotNil)
+
+	expectedDesktopFilePath1 := filepath.Join(dirs.SnapDesktopFilesDir, "org.example.Foo.desktop")
+	c.Assert(osutil.FileExists(expectedDesktopFilePath1), Equals, false)
+	expectedDesktopFilePath2 := filepath.Join(dirs.SnapDesktopFilesDir, "foo_foobar.desktop")
+	c.Assert(osutil.FileExists(expectedDesktopFilePath2), Equals, false)
+
+	// generate .desktop file in the package baseDir
+	baseDir := info.MountDir()
+	c.Assert(os.MkdirAll(filepath.Join(baseDir, "meta", "gui"), 0755), IsNil)
+	c.Assert(os.WriteFile(filepath.Join(baseDir, "meta", "gui", "org.example.Foo.desktop"), mockDesktopFile, 0644), IsNil)
+	c.Assert(os.WriteFile(filepath.Join(baseDir, "meta", "gui", "foobar.desktop"), mockDesktopFile, 0644), IsNil)
+
+	err := wrappers.EnsureSnapDesktopFiles([]*snap.Info{info})
+	c.Assert(err, IsNil)
+
+	for _, expectedDesktopFilePath := range []string{expectedDesktopFilePath1, expectedDesktopFilePath2} {
+		c.Assert(osutil.FileExists(expectedDesktopFilePath2), Equals, true)
```

Comment:
I think `expectedDesktopFilePath2` should just be `expectedDesktopFilePath` here? But this check is already superfluous, as the below `os.Stat` will error if it doesn't exist.

Reply:
Nice catch, will remove the check